modules/sdk/gradle-plugins-js-module-config-generator/README.markdown

```diff
@@ -0,0 +1,115 @@
+plugin is applied:
+
+Property Name | Default Value
+------------- | -------------
+[`moduleConfigFile`](#moduleconfigfile) | `"${project.projectDir}/package.json"`
+[`outputFile`](#outputfile) | `"${sourceSets.main.output.resourcesDir}/META-INF/config.json"`
+[`sourceDir`](#sourcedir) | `"${sourceSets.main.output.resourcesDir}/META-INF/resources"`
+
+The plugin also adds the following dependencies to tasks defined by the `java`
+plugin:
+
+Name | Depends On
+---- | ----------
+`classes` | `configJSModules`
+
+If the [`com.liferay.js.transpiler`](https://github.com/liferay/liferay-portal/tree/master/modules/sdk/gradle-plugins-js-transpiler)
+plugin is applied, the `configJSModules` task is configured to always run after
+the `transpileJS` task.
+
+### ConfigJSModulesTask
+
+Tasks of type `ConfigJSModulesTask` extend `ExecuteNodeScriptTask`, so all its
+properties and methods, such as `args`, `inheritProxy` and `workingDir` are
+available. The `ConfigJSModulesTask` instances also implement the
+[`PatternFilterable`](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/util/PatternFilterable.html)
+interface, which allows to specify include and exclude patterns for the files in
+[`sourceDir`](#sourcedir) to process.
+
+They also have the following properties set by default:
+
+Property Name | Default Value
+------------- | -------------
+[`includes`](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/util/PatternFilterable.html#getIncludes()) | `["**/*.es.js*", "**/*.soy.js*"]`
+`scriptFile` | `"${downloadLiferayModuleConfigGenerator.moduleDir}/bin/index.js"`
+
+Purpose of these tasks is to run the Liferay AMD Module Config Generator from
+the included files in [`sourceDir`](#sourceDir). The Generator will process
+these files and it will create a configuration file in the location specified by
+the [`outputFile`](#outputfile) property.
+
+#### Task Properties
+
+Property Name | Type | Default Value | Description
+------------- | ---- | ------------- | -----------
+`configVariable` | `String` | `null` | The configuration variable to which the modules should be added. It sets the `--config` argument.
+`ignorePath` | `boolean` | `false` | Whether not to create module `path` and `fullPath` properties. It sets the `--ignorePath` argument.
+`keepFileExtension` | `boolean` | `false` | Whether to keep the file extension when generating module name. It sets the `--keepExtension` argument.
+`lowerCase` | `boolean` | `false` | Whether to convert file name to lower case before using it as module name. It sets the `--lowerCase` argument.
+<a name="moduleconfigfile"></a>`moduleConfigFile` | `File` | `null` | The JSON file which contains configuration data for the modules. It sets the `--moduleConfig` argument.
+`moduleExtension` | `String` | `null` | If set, use the provided string as an extension instead to get it automatically from the file name. It sets the `--extension` argument.
```

Comment:
Hi! All these descriptions are shamelessly copied from https://github.com/liferay/liferay-module-config-generator, maybe @jbalsas knows more about it!

And the descriptions in the `README` of `gradle-plugins-js-transpiler` are copied from https://github.com/metal/metal-cli/blob/master/lib/commands/build/options.js.